packages/flutter/lib/src/painting/image_stream.dart

```diff
@@ -396,22 +428,117 @@ abstract class ImageStreamCompleter with Diagnosticable {
     }
   }
 
+  /// Adds a listener callback that is called whenever a new concrete [ImageInfo]
+  /// object is available or an error is reported. If a concrete image is
+  /// already available, or if an error has been already reported, this object
+  /// will notify the listener synchronously.
+  ///
+  /// If the [ImageStreamCompleter] completes multiple images over its lifetime,
+  /// this listener's [ImageStreamListener.onImage] will fire multiple times.
+  ///
+  /// Listeners added by this method are considered to be actively interested
+  /// in decoding and using new frames from multi-frame image streams. It is
+  /// also possible to [addPassiveListener], which will only be called if there
+  /// are other active listeners to drive the stream forward.
+  ///
+  /// {@macro flutter.painting.imageStream.addListener}
+  void addListener(ImageStreamListener listener) {
+    _checkDisposed();
+    _listeners.add(listener);
```

Comment:
nit: rename _listeners to _activeListeners to avoid confusion in the code of what's included here?

Reply:
Done